ee/tabby-webserver/src/service/background_job/mod.rs

```diff
@@ -23,6 +24,7 @@ use self::{
 pub enum BackgroundJobEvent {
     Scheduler(RepositoryConfig),
     SyncThirdPartyRepositories(ID),
+    IndexIssues(ID),
```

Comment:
I recommend creating a new kind of background job, something like`SchedulerGithubRepository` (i don't recommend it being something like `SchedulerThirdPartyRepository`, because not every integrations might have issues / prs concept) - which completes:
1. repository sync + code index building.
2. issue sync + doc index building.

Reply:
Responded in slack, but will reply here too: I chose not to proceed with this since syncing the list of repositories uses an integration ID, and indexing issues uses a repository ID. If we merge these we will lose the precision of being able to index the issues of a single repository when it is switched to activee.